remotes/docker/resolver.go

```diff
@@ -354,6 +354,11 @@ func (r *dockerBase) retryRequest(ctx context.Context, req *http.Request, respon
 			req.Method = http.MethodGet
 			return copyRequest(req)
 		}
+	} else if last.StatusCode == http.StatusForbidden {
+		if r.username != "" && r.secret != "" {
+			r.useBasic = true
```

Comment:
This behavior doesn't exactly make sense. Shouldn't this be included on the first pass? It seems dangerous to just set `useBasic` after a response. Why wasn't this request made credentials in the first place? It seems like needs to analyze the `WWW-Authenticate` header in the response to see if basic credentials are the right approach. Otherwise, this leaks credentials.